drivers/serial/Kconfig.nrfx_uart_instance

```diff
@@ -66,6 +73,43 @@ config UART_$(nrfx_uart_num)_NRF_ASYNC_LOW_POWER
 config UART_$(nrfx_uart_num)_NRF_HW_ASYNC_TIMER
 	int "Timer instance"
 	depends on UART_$(nrfx_uart_num)_NRF_HW_ASYNC
+	depends on UART_NRFX_UARTE_LEGACY_SHIM
+
+config UART_$(nrfx_uart_num)_TX_CACHE_SIZE
+	int "TX cache buffer size"
+	depends on !UART_NRFX_UARTE_LEGACY_SHIM
+	default 8
+	help
+	  For UARTE, TX cache buffer is used when provided TX buffer is not located
+	  in memory which can be used by the EasyDMA.
+
+config UART_$(nrfx_uart_num)_RX_CACHE_SIZE
+	int "RX cache buffer size"
+	depends on !UART_NRFX_UARTE_LEGACY_SHIM
+	default 32 if $(dt_nodelabel_has_compat,ram3x,$(DT_COMPAT_MMIO_SRAM))
+	default 5
+	range 5 255
+	help
+	  For UARTE, RX cache buffer is used when provided RX buffer is not located
+	  in memory which can be used by the EasyDMA. It is also used to store
+	  flushed data.
+
+config UART_$(nrfx_uart_num)_A2I_RX_SIZE
+	depends on !UART_NRFX_UARTE_LEGACY_SHIM
+	int "Asynchronous to interrupt driven adaptation layer RX buffer size"
+	default 64 if UART_$(nrfx_uart_num)_INTERRUPT_DRIVEN
+	default 0
+	help
+	  Amount of space dedicated for RX. It is divided into chunks with some
+	  amount of that space used for control data.
+
+config UART_$(nrfx_uart_num)_A2I_RX_BUF_COUNT
```

Comment:
not a fan is instance-based options, these do not scale well and perpetuate the problem nrfx-based driver have: hardcoded relationship with hw instance numbers. Isn't there a better option?

Reply:
ping